src/BrowserFetcher.ts

```diff
@@ -321,10 +321,43 @@ function install(archivePath: string, folderPath: string): Promise<unknown> {
 }
 
 async function extractZip(zipPath: string, folderPath: string): Promise<void> {
-  try {
-    await extract(zipPath, {dir: folderPath});
-  } catch (error) {
-    return error;
+  const nodeVersion = process.version;
+
+  /* There is currently a bug with extract-zip and Node v14.0.0 that
+   * causes extractZip to silently fail:
+   * https://github.com/puppeteer/puppeteer/issues/5719
+   *
+   * Rather than silenty fail if the user is on Node 14 we instead
+   * detect that and throw an error directing the user to that bug. The
+   * rejection message below is surfaced to the user in the command
+   * line.
+   *
+   * The issue seems to be in streams never resolving so we wrap the
+   * call in a timeout and give it 100ms to resolve before deciding on
+   * an error.
+   *
+   * If the user is on Node < 14 we maintain the behaviour we had before
+   * this patch.
+   */
+  if (nodeVersion.startsWith('v14.')) {
+    await new Promise((resolve, reject) => {
+      let extractedResolved = false;
+      setTimeout(async() => {
+        await extract(zipPath, {dir: folderPath});
+        extractedResolved = true;
+      }, 10);
+
+      if (extractedResolved)
+        resolve();
+      else
+        reject(`Puppeteer currently does not work on Node v14 due to an upstream bug. Please see: https://github.com/puppeteer/puppeteer/issues/5719 for details.`);
+    });
```

Comment:
Current code will always reject.
I think the code should be something like:
```suggestion
    let timeoutReject;
    const timeoutPromise = new Promise((resolve, reject) => { timeoutReject = reject })
    const timeoutToken = setTimeout(
      () => timeoutReject(new Error(`Puppeteer currently does not work on Node v14 due to an upstream bug. Please see: https://github.com/puppeteer/puppeteer/issues/5719 for details.`)),
      10 * 1000 // wait 10 second for 200MB zip to finish
    );
    await Promise.race([
      extract(zipPath, {dir: folderPath}),
      timeoutPromise
    ]);
    clearTimeout(timeoutToken);
```

Reply:
ah nice catch, thanks 👍